### PR TITLE
ENT-4085- add default field in enterprise customer identity provider table

### PR DESCRIPTION
**Pull request by zamanafzal**
Add default field in enterprise customer identity provider table.
Change can be seen locally at the following URL
http://localhost:18000/admin/enterprise/enterprisecustomer/enterprisecustomer_uuid/change

Screenshot 2021-02-08 at 1 57 09 PM

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/models.py [hunk 1]
```diff
@@ -1456,6 +1456,10 @@ class EnterpriseCustomerIdentityProvider(TimeStampedModel):
         unique=True,
         help_text="Slug field containing a unique identifier for the identity provider.",
     )
+    default = models.BooleanField(
```
Comment (sameenfatima78) [inline enterprise/models.py:1459]: Wouldn't we need to make any code changes in the enterprise SSO login flow to make decisions based on this flag?

**Review by sameenfatima78 — APPROVED** [review]
LGTM, just one question

**Comment by HammadAhmadWaqas** [comment]
how would we make sure users would not select more than one IDP as default?

**Comment by zamanafzal** [comment]
@HammadAhmadWaqas This is something that users will not do but the enterprise admin and we need to educate them. But yes we can discuss this in standup tomorrow.

File: enterprise/models.py (hunk 1 above)
Comment (zamanafzal) [reply enterprise/models.py:1459]: @sameenfatima78 That is part of another ticket.

File: enterprise/admin/forms.py [hunk 2]
```diff
@@ -522,6 +523,20 @@ def clean(self):
                 ),
             )
 
+        # There can be only default provider in case an enterprise has multiple providers.
+        if default_provider is True:
```
Comment (HammadAhmadWaqas) [inline enterprise/admin/forms.py:527]: can we just write `if default_provider`

File: enterprise/admin/forms.py [hunk 3]
```diff
@@ -522,6 +523,20 @@ def clean(self):
...
+            default_provider_forms = int(self.data.get('enterprise_customer_identity_providers-TOTAL_FORMS'))
+            default_provider_values = [self.data.get('enterprise_customer_identity_providers-' +
```
Comment (HammadAhmadWaqas) [inline enterprise/admin/forms.py:529]: as list comprehension has been spread to multiple lines should not we use simple for loop?

File: enterprise/admin/forms.py [hunk 4]
```diff
@@ -522,6 +523,20 @@ def clean(self):
...
+                                                    str(i) + '-default_provider')
```
Comment (HammadAhmadWaqas) [inline enterprise/admin/forms.py:530]: can we use a formatted string here for better readability?

File: enterprise/admin/forms.py [hunk 5]
```diff
@@ -522,6 +523,20 @@ def clean(self):
...
+                                      for i in range(default_provider_forms)]
+            providers_marked_default = default_provider_values.count('on')
+            if providers_marked_default>1:
```
Comment (HammadAhmadWaqas) [inline enterprise/admin/forms.py:533]: space is missing from both side of `>` operator

Comment (HammadAhmadWaqas) [reply enterprise/admin/forms.py:533]: after updating this snippet will look like this.
```
        if default_provider:
            identity_provider_total_forms = int(self.data.get('enterprise_customer_identity_providers-TOTAL_FORMS'))
            default_provider_values = []
            for form_num in range(identity_provider_total_forms):
                default_provider_values.append(
                    self.data.get(f'enterprise_customer_identity_providers-{form_num}-default_provider')
                )
            providers_marked_default = default_provider_values.count('on')
            if providers_marked_default > 1:
                raise ValidationError(
                    {
                        'Please select only one default provider.'
                    }
                )
```

**Review by HammadAhmadWaqas — APPROVED** [review]
your code is working, a few small changes if you consider it right we can make.

File: enterprise/admin/forms.py (hunk 3 above)
Comment (zamanafzal) [reply enterprise/admin/forms.py:529]: In that case, we need to create an empty list and then append the data in that. I think it looks cleaner than that solution.

**Pull request merged**